### PR TITLE
Add blorg recipe

### PR DESCRIPTION
**Pull request by guilhermecomum**
### Brief summary of what the package does

Generate Static HTML websites from Org-Mode files.

The blorg an Emacs-Lisp API rather than a command line tool. The use of the API becomes the description of a pipeline that takes lists of Org-Mode files as input, templatize them and generate HTML files as an output.
### Direct link to the package repository

https://github.com/emacs-love/blorg
### Your association with the package

A contributor

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

**Comment by riscy** [comment]
Unfortunately there is a [very old package with the name blorg](https://github.com/emacsattic/blorg) already in the attic; it will be less confusing for users if you choose a different name, although I realize blorg has a certain ring to it. :)  

In the meantime, some very quick checks --

- blorg.el#L823: It's safer to sharp-quote function names; use `#'`
- blorg.el#L753: It's safer to sharp-quote function names; use `#'`
- blorg.el#L758: It's safer to sharp-quote function names; use `#'`
- blorg.el#L670: Use `when ...` instead of `unless (null ...)`
- blorg.el#L100: Prefer `https` over `http` if possible ([why?](https://news.ycombinator.com/item?id=22933774))

**Comment by conao3** [comment]
emacsattic is maintained by @tarsius, so I just mention him to make him know this PR.

**Comment by clarete** [comment]
Hi @riscy! Thank you for the review. We've updated the code to fix the issues you caught. The last warning is a bit trickier though. The URL at that line contains the default URL that is intended to be used with the `localhost` domain.
I think it'd be a bit less ergonomic to use it locally if people had to setup a more secure solution just for local tests. Do you think it's OK if we leave it like that?

Also, we're thinking of a good name. feel free to chime in and suggest one if anything comes to mind!

**Comment by riscy** [comment]
> Do you think it's OK if we leave it like that?

Oh darn, that was a false positive -- localhost is fine over http of course.  I should clean that up.

I think if I saw "weblorg" I would know what this is right away, or org-blog.  There are also a couple packages on MELPA that do [something similar](https://melpa.org/#/?q=org%20blog) such as [org-static-blog](https://melpa.org/#/org-static-blog) which looks actively maintained and probably warrants a comparison.

**Comment by clarete** [comment]
> > Do you think it's OK if we leave it like that?
> 
> Oh darn, that was a false positive -- localhost is fine over http of course. I should clean that up.

Nice, thanks for confirming! 😄 


> I think if I saw "weblorg" I would know what this is right away, or org-blog.

That is rather a great name, I like how one can see it either as "WeBlorg" or "Web Lorg"!!
 @guilhermecomum what do you think? 😄 

>  There are also a couple packages on MELPA that do [something similar](https://melpa.org/#/?q=org%20blog) such as [org-static-blog](https://melpa.org/#/org-static-blog) which looks actively maintained and probably warrants a comparison.

yeah, I looked at most of them, and org-static-blog was definitely the closest to what I wanted before starting this new project.  The main missing features for me were:

 0. I didn't want to concatenate strings to make templates. (Ended up building https://melpa.org/#/templatel to solve that) 
 1. Wanted a clean way to define routes and link them in a way that would be easy to find broken links (this new guy we've been working on has the `url_for` mechanism that is plugged to the routes declared in a `publish.el`  file
 2.  wanted an easy way to build category pages and archive pages, which is easily achievable by passing an `:input-aggregator` function to a given route.
 3. Didn't want to depend on software  outside the Emacs ecosystem (like wordpress or google blogger)

File: recipes/blorg [hunk 1]
```diff
@@ -0,0 +1 @@
+(blorg :fetcher github :branch "main" :repo "emacs-love/blorg")
```
Comment (riscy) [inline recipes/blorg:1]: Just updating the recipe here.  I assume `main` is your default branch, in which case it doesn't need to be specified:
```suggestion
(weblorg :fetcher github :repo "emacs-love/weblorg")
```

**Comment by riscy** [comment]
Taking a closer look at the source now -- just a couple of small issues I see:

- The semicolon at the end of your summary is unnecessary (so your summary line should be `;;; weblorg.el --- Static site generator for org-mode -*- lexical-binding: t -*-`, i.e. no semicolon before lexical binding)
- Should `weblorg-module-dir` be a `defconst` instead of a `defvar`?  Also, sometimes `load-file-name` is nil which will cause this declaration to fail, such as when a user is running <kbd>M-x eval-buffer</kbd>; here's how e.g. Emacsql handles it: `(defvar emacsql-data-root (file-name-directory (or load-file-name buffer-file-name))`

Minor:
- weblorg.el#L605: This double negation could be collapsed (`not` is an alias for `null`)
- weblorg.el#L661: This double negation could be collapsed (`not` is an alias for `null`)
- weblorgs source code -> weblorg's source code
- Genenrate -> Generate
- enviroment -> environment
- writtend directly -> written directly
- circiut -> circuit

**Comment by clarete** [comment]
Hi @riscy 👋🏾 thank you so much for all the time you spent looking into this and for this very thorough review! I just pushed fixes for the things you pointed out 🙇🏾

**Comment by clarete** [comment]
one thing that I initially overlooked is the shipping of assets, like HTML templates and built CSS files.  I don't think the themes we currently have there is really ready to be packaged anyway.  I think it shouldn't block this PR, but I'll make sure I properly document that for now themes aren't being shipped. Here's the issue for that: https://github.com/emacs-love/weblorg/issues/17.

we'll exclude sources from built files and the built files will be minified and packed tightly to keep the overall package size as small as possible. We'll consider maybe only copying the templates and providing a `cdn` option for assets, so maybe people can pick it from github pages for example. so we don't have to package anything but HTML templates (which are quite few and small).

I'll look for good practices on packaging assets for melpa but if you have any pointers, it will be very much appreciated!

once again, thank you for all the work and fun 🙇🏾

**Comment by riscy** [comment]
> I'll look for good practices on packaging assets for melpa but if you have any pointers, it will be very much appreciated!

One thing to be aware of is that even if your recipe excludes large asset files, an installer like straight.el will pull the entire repository down.  This looks good to go.

**Pull request merged**